server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -254,6 +323,31 @@ public void onFailure(Exception e) {
         }
     };
 
+    /**
+     * Handle the setting (and possible creation) of APM metrics
+     */
+    // default visibility for testing purposes
+    void handleMetrics(List<HealthIndicatorResult> healthIndicatorResults) {
+        if (healthIndicatorResults != null) {
+            for (HealthIndicatorResult result : healthIndicatorResults) {
+                String metricName = result.name();
+                LongGaugeMetric metric = this.metrics.get(metricName);
+                if (metric == null) {
+                    metric = LongGaugeMetric.create(
```

Comment:
@pgomulka I had that approach going and changed it in [this commit](https://github.com/elastic/elasticsearch/pull/102765/commits/4ebbb03836a0d6b9fb3249865bea321cf58e7680). I figured that it would be a maintenance burden to update that list as we add new indicators so I went with this create-when-first-seen approach. It's similar to what we do with the textual logging feature of this class, in that we log anything we see. I'll be interested in what @dakrone thinks about either of the approaches here.